rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -315,6 +320,21 @@ private Response launch(String yaml, EntitySpec<? extends Application> spec) {
         }
     }
 
+    private void waitForStart(final Application app, Duration timeout) {
+        // without this UI shows "stopped" sometimes esp if brooklyn has just started,
+        // because app goes to stopped state if it sees unstarted children,
+        // and that gets returned too quickly, before the start has taken effect
```

Comment:
This feels like a treatment of the symptom rather than the cause (which is probably rather more involved).